pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go

```diff
@@ -36,5 +36,5 @@ func SetMapValue(attrs pcommon.Map, mapKey string, val interface{}) {
 	}
 
 	SetValue(value, val)
-	attrs.Upsert(mapKey, value)
+	value.CopyTo(attrs.UpsertEmpty(mapKey))
```

Comment:
SetValue(attrs.UpsertEmpty(mapKey), val)

Should be a separate PR.

Reply:
Yeah I wanted to keep this PR's scope limited and apply other improvements after